sw/host/provisioning/ft_lib/src/lib.rs

```diff
@@ -548,26 +549,39 @@ pub fn check_rom_ext_boot_up(
             .map(|s| s.as_str())
             .unwrap_or("unknown"),
     );
+    let t0 = Instant::now();
 
-    // Timeout for waiting for a potential error message indicating invalid UDS certificate.
-    // This value is tested on fpga cw340 and could be potentially fine-tuned.
-    const UDS_CERT_INVALID_TIMEOUT: Duration = Duration::from_millis(200);
+    // Timeout for waiting for a potential error message indicating invalid UDS
+    // certificate or for the Owner firmware successful startup.
+    //
+    // These values were tested on fpga cw340 and could be potentially fine-tuned.
+    let slot_b_startup_timeout: Duration =
+        Duration::from_millis(if owner_fw_success_string.is_none() {
+            200
+        } else {
+            1500
+        });
 
-    let t0 = Instant::now();
-    let result = UartConsole::wait_for(
-        &*uart_console,
-        r".*UDS certificate not valid.",
-        UDS_CERT_INVALID_TIMEOUT,
-    );
     response.stats.log_elapsed_time("rom_ext-done", t0);
+    let fw_failure_anchor = r"Invalid UDS certificate detected!";
```

Comment:
nit (it is a ROM_EXT failure message):
```suggestion
    let rom_ext_failure_msg = r"Invalid UDS certificate detected!";
```

Reply:
done